boot/bootutil/src/loader.c

```diff
@@ -185,11 +201,17 @@ boot_status_source(void)
              table->bst_magic_scratch   == state_scratch.magic) &&
             (table->bst_copy_done_slot0 == 0    ||
              table->bst_copy_done_slot0 == state_slot0.copy_done)) {
-
-            return table->bst_status_source;
+            uint8_t source = table->bst_status_source;
```

Comment:
This prompts an interesting dilemma - I guess we need some sort of coding style for mcuboot.  The Mynewt coding style (https://github.com/apache/incubator-mynewt-core/blob/master/CODING_STANDARDS.md) differs from Zephyr's.  Notably, in Mynewt:
* Indentation is done with four spaces (no tabs)
* No mixing of variable declarations and statements within a block.

I don't personally mind which style we settle on, but it would be good to be consistent.  Since most of the mcuboot code is already written in the Mynewt style, it might be easiest to stick with that one (but like I said, I don't have a strong opinion).